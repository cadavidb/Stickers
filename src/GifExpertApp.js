import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = ['kermit memes'] }) => {
    

    const [categories, setCategories] = useState( defaultCategories );

    return (
        <>
            <h2 className="animate__animated animate__heartBeat">Gifs animados</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

            <ol>
                {
                    categories.map( category  => (
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    ))
                }
            </ol>

        </>
    )
}



