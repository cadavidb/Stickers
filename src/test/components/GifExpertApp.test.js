import React from 'react'
import shallow from 'enzyme/build/shallow'
import { GifExpertApp } from '../../GifExpertApp'

describe('Pruebas en <GifExpertApp>', () => {
    test('debe hacer match con el snapshot', () => {

        const wrapper=shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();

    })

    test('debe mostrar una lista de categorias', () => {
        const categories=['spider man','cops'];
        const wrapper=shallow(<GifExpertApp defaultCategories={categories}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
})
