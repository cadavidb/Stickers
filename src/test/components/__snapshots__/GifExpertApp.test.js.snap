// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp> debe hacer match con el snapshot 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__heartBeat"
  >
    Gifs animados
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="memes"
      key="memes"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__heartBeat"
  >
    Gifs animados
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="spider man"
      key="spider man"
    />
    <GifGrid
      category="cops"
      key="cops"
    />
  </ol>
</Fragment>
`;
